File: catamesh-core/src/main/java/dev/catamesh/core/model/DiffResult.java
package dev.catamesh.core.model;

import java.util.List;

public final class DiffResult {
    private final DiffTreeNode root;
    public DiffResult(DiffTreeNode root) {
        this.root = root;
    }

    public DiffTreeNode getRoot() {
        return root;
    }

    public DiffSummary getSummary() {
        return root.computeSummary();
    }

    public boolean hasChanges() {
        return getSummary().totalChanges() > 0;
    }
}

File: catamesh-cli/src/core/model/DiffResult.ts
import {DiffSummary} from "./DiffSummary";
import {DiffTreeNode} from "./DiffTreeNode";
import {PolicyRule} from "./PolicyRule";

export interface DiffResult {
    root: DiffTreeNode;
    summary: DiffSummary;
    policyRules?: PolicyRule[] | null;
}
